File: app_manager/appmanager/src/main/java/app/editors/manager/managers/retrofit/BaseInterceptor.java
package app.editors.manager.managers.retrofit;

import androidx.annotation.NonNull;

import java.io.IOException;

import app.editors.manager.app.Api;
import app.editors.manager.app.App;
import app.editors.manager.managers.exceptions.NoConnectivityException;
import app.editors.manager.managers.tools.PreferenceTool;
import lib.toolkit.base.managers.utils.NetworkUtils;
import lib.toolkit.base.managers.utils.StringUtils;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class BaseInterceptor implements Interceptor {

    private static final String KEY_AUTH = "Bearer ";

    private PreferenceTool mPreferenceTool;

    public BaseInterceptor() {
        mPreferenceTool = App.getApp().getAppComponent().getPreference();
    }

    @NonNull
    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        checkConnection();

        Request request = chain.request();
        int version = StringUtils.convertServerVersion(mPreferenceTool.getServerVersion());

        if (mPreferenceTool.getSslState() && version >= 10 && mPreferenceTool.getToken() != null) {
            request = request.newBuilder().header(Api.HEADER_AUTHORIZATION, KEY_AUTH +
                    request.header(Api.HEADER_AUTHORIZATION)).build();
        }

        return chain.proceed(request);
    }

    private void checkConnection() throws NoConnectivityException {
        if (!NetworkUtils.isOnline(App.getApp())) {
            throw new NoConnectivityException();
        }
    }

}

File: app_manager/appmanager/src/main/java/app/editors/manager/managers/retrofit/BaseInterceptor.kt
package app.editors.manager.managers.retrofit

import android.os.Build
import app.documents.core.network.ApiContract
import app.editors.manager.BuildConfig
import app.editors.manager.app.App
import app.editors.manager.managers.exceptions.NoConnectivityException
import lib.toolkit.base.managers.utils.NetworkUtils.isOnline
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class BaseInterceptor(val token: String?) : Interceptor {

    companion object {
        private const val KEY_AUTH = "Bearer "
    }

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        checkConnection()
        val token = if (chain.request().url().host().contains(ApiContract.PERSONAL_HOST)) {
            token
        } else {
            KEY_AUTH + token
        }
        return chain.proceed(
            chain.request().newBuilder().addHeader(
                ApiContract.HEADER_AUTHORIZATION,
                token ?: ""
            )
                .addHeader(
                    ApiContract.HEADER_AGENT,
                    "Android ONLYOFFICE Documents (id = ${BuildConfig.APPLICATION_ID}, SDK = ${Build.VERSION.SDK_INT}, build = ${BuildConfig.VERSION_CODE}, appName = ${BuildConfig.VERSION_NAME}"
                )
                .build()
        )
    }

    @Throws(NoConnectivityException::class)
    private fun checkConnection() {
        if (!isOnline(App.getApp())) {
            throw NoConnectivityException()
        }
    }
}